src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -518,6 +529,8 @@ public static ConsumeCodeResponse consumeCode(TenantIdentifier tenantIdentifier,
                 }
             }
 
+            // We do not need this cleanup if we are creating the user, since it uses the email/phoneNumber of the
+            // device, which has already been cleaned up
```

Comment:
this comment says we do not need this cleanuip, but still there is some cleanup happening below. So it's confusing. Also, why don't we need the cleanup? Here we are deleting the other codes that belong to the email / phone number

Reply:
this is an old comment, I just moved it to the relevant place. anyway, it is talking about the other branch in the if that does not need clean up, I have rephrased it to make it clear.